src/renderers/shared/fiber/ReactChildFiber.js

```diff
@@ -62,6 +64,24 @@ const {
   Deletion,
 } = ReactTypeOfSideEffect;
 
+function transferRef(fiber: Fiber, element: ReactElement<any>) {
+  if (typeof element.ref === 'string' && element._owner) {
+    // Should we check if this string refs is equal to the previous string ref
+    // and reuse it?
+    const ownerFiber : ?Fiber = (element._owner : any);
```

Comment:
This is not guaranteed to be a Fiber and it is not guaranteed to be a `ClassComponent` since you attach owners to functional components too.